src/common/fs/file.h

```diff
@@ -0,0 +1,448 @@
+// Copyright 2021 yuzu Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+#include <span>
+#include <type_traits>
+#include <vector>
+
+#include "common/concepts.h"
+#include "common/fs/fs_types.h"
+#include "common/fs/fs_util.h"
+
+namespace Common::FS {
+
+enum class SeekOrigin {
+    /// Seeks from the start of the file.
+    SetOrigin,
+    /// Seeks from the current file pointer position.
+    CurrentPosition,
+    /// Seeks from the end of the file.
+    End,
+};
+
+/**
+ * Opens a file stream at path with a specified open mode.
+ *
+ * @param file_stream Reference to file stream
+ * @param path Filesystem path
+ * @param open_mode File stream open mode
+ */
+template <typename FileStream>
+void OpenFileStream(FileStream& file_stream, const std::filesystem::path& path,
+                    std::ios_base::openmode open_mode) {
+    file_stream.open(path, open_mode);
+}
+
+template <typename FileStream, typename Path>
+void OpenFileStream(FileStream& file_stream, Path path, std::ios_base::openmode open_mode) {
```

Comment:
```suggestion
void OpenFileStream(FileStream& file_stream, const Path& path, std::ios_base::openmode open_mode) {
```

Ditto where applicable in the file